### PR TITLE
Tweak test slicing

### PR DESCRIPTION
**Pull request by keeplearningandtrying**
Few minor tweak and polish for unit test

### CODE REVIEW

File: src/test/java/com/example/external/data/CarRepositoryIntegrationTests.java [hunk 1]
```diff
@@ -30,7 +30,9 @@
  * @author Mark Paluch
  */
 @RunWith(SpringJUnit4ClassRunner.class)
-@SpringBootTest
+// no need to load the full context
+//@SpringBootTest
+@DataJpaTest
```
Comment (mp911de) [inline src/test/java/com/example/external/data/CarRepositoryIntegrationTests.java]: Thanks a lot. Can we remove the commented code?

File: src/test/java/com/example/web/CarControllerRestTests.java [hunk 2]
```diff
@@ -48,7 +48,8 @@
     @Autowired
     private TestRestTemplate restTemplate;
 
-    @MockBean
+    // As spring will configure the full context, no need to mock this bean
+    //@MockBean
```
Comment (mp911de) [inline src/test/java/com/example/web/CarControllerRestTests.java]: This test is intended to show Mockito usage with Spring so `@MockBean` is present on purpose.

File: src/test/resources/application.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 spring.jackson.serialization.indent_output=true
 spring.jpa.hibernate.ddl-auto=none
-spring.datasource.url=jdbc:mysql://localhost/springboot
-spring.datasource.username=springboot
-spring.datasource.password=springboot
+#spring.datasource.url=jdbc:mysql://localhost/springboot
```
Comment (mp911de) [inline src/test/resources/application.properties]: Let's remove the commented code.

**Review by mp911de — CHANGES_REQUESTED** [review]
Thanks a lot for your pull request and the found issue with `@DataJpaTest`. I added a few comments. Care to revisit these?

File: src/test/java/com/example/external/data/CarRepositoryIntegrationTests.java (hunk 1 above)
Comment (keeplearningandtrying) [reply src/test/java/com/example/external/data/CarRepositoryIntegrationTests.java]: Yes. Thanks.

File: src/test/java/com/example/web/CarControllerRestTests.java (hunk 2 above)
Comment (keeplearningandtrying) [reply src/test/java/com/example/web/CarControllerRestTests.java]: Got you.

**Comment by mp911de** [comment]
You could have left this pull request open. GitHub picks up any changes that you do on your fork/branch.

**Pull request closed without merging**